### ISSUE TITLE
Use compiled binary on machines with avconv?

### ISSUE THREAD

**Issue by Lesik** [qTox/qTox#2776]
The precompiled binary available [here](https://build.tox.chat/view/Clients/job/qTox_build_linux_x86-64_release/lastSuccessfulBuild/artifact/qTox_build_linux_x86-64_release.tar.xz) crashes on Ubuntu 14.04 and Trisquel 7.0 Belenos. I presume that this is due to Ubuntu 14.04 using `avconv` instead of `ffmpeg`. I have compiled `ffmpeg`, but the precompiled binary still crashes (`qtox: src/audio/audio.cpp:752: void Audio::subscribeOutput(Audio::SID&): Zusicherung »sid« nicht erfüllt.`).

![screenshot](https://i.imgur.com/XaeHx21.png)

The system is Trisquel 7.0 Belenos (based on Ubuntu 14.04).

Am I missing something? Is there a way to make the binary work with a self-compiled ffmpeg?

**Comment by zetok** [comment]
> I have compiled ffmpeg, but the precompiled binary still crashes
> 
> Is there a way to make the binary work with a self-compiled ffmpeg?

You would want to use `LD_LIBRARY_PATH` variable - i.e. point it to the place where compiled ffmpeg `.so` are located, and you could launch qTox;
e.g.

```
export LD_LIBRARY_PATH="/path/to/compiled/ffmpeg/so/files/"
./qtox
```

If you're not quite sure how to use it, there are quite a few explanations on the internet :)

---

@Zer0-One why is packaging broken?

**Comment by Zer0-One** [comment]
@zetok that's a question for @tux3 , I don't maintain qTox's builds.

**Comment by Lesik** [comment]
Thanks.

**Comment by antis81** [comment]
@Lesik Much more simple: Select a valid audio device in qTox A/V settings and the crash will go away. :)

**Comment by tux3** [comment]
@antis81 
I haven't looked into this, but that sounds like a bug, no? If there's a crash, it has to be fixed.

**Comment by zetok** [comment]
@tux3 yeah, it's fixed in #2785

**Comment by antis81** [comment]
@tux3 It's an assertion because, when a call is made with an invalid in/out audio device it is not possible to create an audio source - obviously. The fix in #2785 includes 2 important changes:
1. A "none" audio device is marked as "not initialized" (Audio::initInput/-Output both return `false` in that case).
2. `CoreAV` does not any longer try to send audio data to an invalid audio source -> this was visible as "spam" in log file.

IMO the "call" and "mute" buttons should be disabled, when explicitly no audio device is selected.

**Comment by tux3** [comment]
Sounds good, thanks.
